hy/lex/parser.py

```diff
@@ -44,7 +44,8 @@ def unicode_char_to_hex(uchr):
         # Covert a unicode char to hex string, without prefix
         return uchr.encode('unicode-escape').decode('utf-8').lstrip('\\U').lstrip('\\u').lstrip('0')
 
-    assert s
+    if not s:
+        return ""
```

Comment:
Is this line (and the corresponding one in `unmangle`) returning the same type as the rest of the function?

Reply:
Yeah, so I don't mind putting an error back in (probably should raise a `ValueError` though) and handling the edge case inside the completer.

Reply:
That sounds reasonable to me.